tests/prague/eip2537_bls_12_381_precompiles/test_bls12_pairing.py

```diff
@@ -235,6 +284,57 @@ def test_invalid(
     )
 
 
+@pytest.mark.slow
+@pytest.mark.parametrize("precompile_gas", [None], ids=[""])
+@pytest.mark.parametrize("expected_output", [Spec.INVALID], ids=[""])
+def test_invalid_multi_inf(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    call_contract_address: Address,
+    sender: EOA,
+    fork: Fork,
+    post: dict,
+):
+    """
+    Test maximum input given the current environment gas limit for the BLS12_PAIRING
+    precompile and an invalid tail.
+    """
+    intrinsic_gas_cost_calculator = fork.transaction_intrinsic_cost_calculator()
+    memory_expansion_gas_calculator = fork.memory_expansion_gas_calculator()
+    extra_gas = 100_000
+
+    environment_gas_limit = Environment().gas_limit
+
+    inf_data = Spec.INF_G1 + Spec.INF_G2
+    input_data = PointG1(Spec.P, 0) + Spec.INF_G2
+
+    while True:
+        precompile_gas = pairing_gas(len(input_data + inf_data))
+        new_tx_gas_limit = (
+            extra_gas
+            + intrinsic_gas_cost_calculator(calldata=input_data + inf_data)
+            + memory_expansion_gas_calculator(new_bytes=len(input_data + inf_data))
+            + precompile_gas
+        )
+        if new_tx_gas_limit > environment_gas_limit:
+            break
+        tx_gas_limit = new_tx_gas_limit
+        input_data = inf_data + input_data
```

Comment:
Should this be extracted so can be reused also in the prev test to avoid repetition? Or maybe it is fine, you're the boss :smile:

Reply:
Yeah, it was simply done in a hurry 😛 

Should be ok.